src/network/ssl_wrapper.hpp

```diff
@@ -359,25 +360,33 @@ class Ssl {
 #else
   inline static std::unordered_set<std::string> GetLocalIpList() {
     std::unordered_set<std::string> ip_list;
-    struct ifaddrs * ifAddrStruct = NULL;
-    struct ifaddrs * ifa = NULL;
-    void * tmpAddrPtr = NULL;
+    int SockFD;
+    SockFD = socket(AF_INET, SOCK_DGRAM, 0);
+    struct ifreq *ifr, *ifend;
+    struct ifreq ifreq;
+    struct ifconf ifc;
+    memset(&ifc, 1, sizeof(ifc));
+    ifc.ifc_req = NULL;
+    if (ioctl(SockFD, SIOCGIFCONF, &ifc) < 0) {
+        printf("ioctl(SIOCGIFCONF): %m\n");
```

Comment:
use Log like [here](https://github.com/analytics-zoo/LightGBM/blob/9f89b04d47ec7ea4f1ceee299db124cd89058564/src/network/ssl_wrapper.hpp#L126) instead of printf